### ISSUE TITLE
Change /web to /public_html

### ISSUE THREAD

**Issue by ghost** [contao/installation-bundle#40]
<a href="https://github.com/t-me"><img src="https://avatars.githubusercontent.com/u/14944366?v=3" align="left" width="42" height="42"></img></a> [Issue](https://github.com/contao/standard-edition/issues/51) by @t-me
January 19th, 2017, 21:15 GMT

Hello All,

I want to change the "/web" directory to "/public_html".

When i follow this link : [symfony.com/override-the-web-directory](http://symfony.com/doc/current/configuration/override_dir_structure.html#override-the-web-directory)

and do a composer update it looks fine. But when i go to install.php i get a error 500. in the log files i see this 

```
[Thu Jan 19 21:06:15.625264 2017] [lsapi:error] [pid 234676] [client 82.75.164.96:0] [host t-me.nl] Backend fatal error: PHP Fatal error:  Uncaught InvalidArgumentException: The target directory "/home/debxxx/domains/t-me.nl/app/../web" does not exist. in /home/debxxx/domains/t-me.nl/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php:85\nStack trace:\n#0 /home/debxxx/domains/t-me.nl/vendor/symfony/symfony/src/Symfony/Component/Console/Command/Command.php(262): Symfony\\Bundle\\FrameworkBundle\\Command\\AssetsInstallCommand->execute(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\BufferedOutput))\n#1 /home/debxxx/domains/t-me.nl/vendor/contao/installation-bundle/src/Controller/InstallationController.php(170): Symfony\\Component\\Console\\Command\\Command->run(Object(Symfony\\Component\\Console\\Input\\ArgvInput), Object(Symfony\\Component\\Console\\Output\\BufferedOutput))\n#2 /home/debxxx/domains/t-me.nl/vendor/contao/installation-bundle/src/Controller/InstallationController.php(133): Contao\\InstallationBundle\\Controller\\InstallationController->runCommand(Object(Symfon in /home/debxxx/domains/t-me.nl/vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Command/AssetsInstallCommand.php on line 85\n
```

The reason i want to change it is because i cant change public_html to web directory as root folder.

**Comment by leofeyer** [comment]
I doubt that this is possible in Contao 4.3. We have made quite some efforts to support a custom directory structure in version 4.4, but there are still a lot of hard corded references to `/web`, which we cannot change without knowing the path to the web folder in Symfony. And by design, Symfony does not know it (only Composer does).

@contao/developers /cc

**Comment by Toflar** [comment]
> The reason i want to change it is because i cant change public_html to web directory as root folder.

Imho this is simply a platform requirement just like the php version etc.

**Comment by fritzmg** [comment]
A customizable web folder is no unusual requirement though.

**Comment by leofeyer** [comment]
> The reason i want to change it is because i cant change public_html to web directory as root folder.

@t-me Actually you are supposed to put everything into the `public_html` folder and then route your domain in cPanel (or whatever hosting control panel there is) to the `web` subfolder.

> A customizable web folder is no unusual requirement though.

Correct. That's why we are trying to support it in version 4.4. But it is not trivial (see above).

**Comment by t-me** [comment]
@leofeyer Thanks for your comment on putting everything in the 'public_html'
I have DirectAdmin as control pannel but can not change the virtualhost file.

I found another solution it is not ideal but it does the trick.

[Found it here...](https://github.com/roots/bedrock/issues/180#issuecomment-104237989)

	Options -Indexes
	# .htaccess primary domain to web redirect
	# this file must be in the (public_html | web ) folder of your hosting account 
	# make the changes to the file according to the instructions.

	<IfModule mod_rewrite.c>
		# Do not change this line.
		RewriteEngine on

		# Uncomment the following lines to add "www." to the domain:
		# RewriteCond %{HTTP_HOST} ^[^.]+\.[^.]+$
		# RewriteRule ^(.*)$ http://www.%{HTTP_HOST}/$1 [L,R=301]
  
		# Uncomment the following lines to Redirect to SSL
		# RewriteCond %{HTTPS} !=on
		# RewriteRule ^(.*)$ https://%{HTTP_HOST}%{REQUEST_URI} [L,R=301]	
	
		# Change %host% to be your primary domain.
		RewriteCond %{HTTP_HOST} ^(www.)?%host%$

		# Change 'web' to be the folder you will use for your primary domain.
		RewriteCond %{REQUEST_URI} !^/web/

		#********************
		# NOT RRCOMMENDED   *
		#********************
		#
		# Uncomment those two lines if you want dirs and files to be accessible out of the
		# your primary domain dir.

		#RewriteCond %{REQUEST_FILENAME} !-f 
		#RewriteCond %{REQUEST_FILENAME} !-d

		# Change 'web' to be the folder you will use for your primary domain.
		RewriteRule ^(.*)$ /web/$1

		# Change %host% to be your primary domain again. 
		# Change 'web' to be the folder you will use for your primary domain 
		# followed by / then the main file for your site, index.php, index.html, etc.
		RewriteCond %{HTTP_HOST} ^(www.)?%host%$ 
		RewriteRule ^(/)?$ web/app.php [L]
	</IfModule>

Hope this will help for other people having the same issue as i have.

_Edited the `.htaccess` with "www" and "ssl" option_

**Comment by leofeyer** [comment]
I have found hard coded references to `/web` in at least the following files:

 * `src/Resources/contao/library/Contao/Combiner.php`
 * `src/EventListener/InsecureInstallationListener.php`
 * `src/Resources/contao/library/Contao/Image.php`
 * `src/Resources/contao/library/Contao/Automator.php`
 * `src/Resources/contao/controllers/FrontendCron.php`

The news and calendar bundles also write into the `web/share` folder, therefore we need to store the path to the web directory somewhere so we can access it in Contao. I understand that this is against the Symfony concept, but until all the resources above have been refactored, I don't think we have a choice.

@contao/developers /cc

**Comment by leofeyer** [comment]
As discussed at the developer's meeting, the parameter shall be named `contao.web_dir`.

**Comment by leofeyer** [comment]
Added the parameter in contao/core-bundle@b307ac43797f6ec7a1036e5c529edcf975aed1ac as basis for the upcoming pull requests.

**Comment by leofeyer** [comment]
@t-me BTW, the easiest solution probably would have been a symlink from `public_html` to `web`.

**Comment by leofeyer** [comment]
Fixed in contao/core-bundle@8a74bb16f659e5c79b6c428c33360dc3f42264a6.
